app/models/null_user.rb

```diff
@@ -0,0 +1,13 @@
+class NullUser
+  def method_missing(method, *args)
+    if respond_to? method
+      nil
+    else
+      super
+    end
+  end
+
+  def respond_to_missing?(*_args)
+    true
+  end
```

Comment:
I usually prefer Null objects that are more specific to the interface they are replacing. 
If you remember from Confident Ruby book, Avdi talks about using a very generic null object can be a black hole that should be used sparingly.

Reply:
I did read Avdi again and had implemented a user interface specific nulluser but then Rubocop got all upset about method_missing needing a call to super and needing respond_to_missing so I went searching elsewhere - it was pretty frustrating. If I don't need a nullobject pattern here I will happily delete the branch lol.

Reply:
If the majority of the interaction needs the user to be logged in, then I wouldn't worry about a null user just yet

Reply:
Sounds good to me! It was an interesting exercise at any rate.